src/main/java/com/github/elmo675/controller/DiaryEntryController.java

```diff
@@ -13,51 +18,119 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.text.ParseException;
+import java.util.stream.Collectors;
 
 @RestController
 @RequestMapping("/api/v1")
-
 public class DiaryEntryController {
 
     @Autowired
     private DiaryEntryRepository DiaryEntryRepository;
 
+    @Autowired
+    ModelMapper modelMapper = new ModelMapper();
+
+    private DiaryEntryResponse convertDiaryEntryToResponse(DiaryEntry diaryEntry) {
+        DiaryEntryResponse DTO = modelMapper.map(diaryEntry, DiaryEntryResponse.class);
+        DTO.setId(diaryEntry.getId());
+        DTO.setAcces(diaryEntry.getAccess());
+        DTO.setContent(diaryEntry.getContent());
+        DTO.setCreatedBy(diaryEntry.getCreatedBy());
+        DTO.setCreatedAt(diaryEntry.getCreatedAt());
+        DTO.setUpdatedAt(diaryEntry.getUpdatedAt());
+        DTO.setUpdatedBy(diaryEntry.getUpdatedBy());
+        return DTO;
+    }
+
+    private DiaryEntryRequest convertDiaryEntryToRequest(DiaryEntry diaryEntry) {
+        DiaryEntryRequest DTO = modelMapper.map(diaryEntry, DiaryEntryRequest.class);
+        DTO.setAccess(diaryEntry.getAccess());
+        DTO.setContent(diaryEntry.getContent());
+        DTO.setCreatedBy(diaryEntry.getCreatedBy());
+        return DTO;
+    }
+
+    private DiaryEntry convertResponseToDiaryEntry(DiaryEntryResponse diaryEntryResponse) throws ParseException {
+        DiaryEntry diaryEntry = modelMapper.map(diaryEntryResponse, DiaryEntry.class);
+        diaryEntry.setId(diaryEntryResponse.getId());
+        diaryEntry.setContent(diaryEntryResponse.getContent());
+        diaryEntry.setAccess(diaryEntryResponse.getAcces());
+        diaryEntry.setUpdatedBy(diaryEntryResponse.getUpdatedBy());
+        diaryEntry.setUpdatedAt(diaryEntryResponse.getUpdatedAt());
+        diaryEntry.setCreatedAt(diaryEntryResponse.getCreatedAt());
+        diaryEntry.setCreatedBy(diaryEntryResponse.getCreatedBy());
+        return diaryEntry;
+    }
+    private DiaryEntry convertRequestToDiaryEntry(DiaryEntryRequest diaryEntryRequest) throws ParseException {
+        DiaryEntry diaryEntry = modelMapper.map(diaryEntryRequest, DiaryEntry.class);
+        //DiaryEntry diaryEntry = new DiaryEntry();
+        diaryEntry.setContent(diaryEntryRequest.getContent());
+        diaryEntry.setAccess(diaryEntryRequest.getAccess());
+        if(diaryEntry.getCreatedBy() != null){
```

Comment:
1. W javie stawia się spacje przed i po nawiasie
2. Stosuje się klamry w tej samej linii co else
3. Stostuje się spacje przed i po klamrze - w tym i poprzednich punktach można użyć opcji auto formatowania w idei
4. Dziwna ta logika if/else. W jakiej sytuacji createdBy jest ustawione? Przecież wcześniej przepisujesz je z request za pomocą mappera, a w request nie ma tego pola. Masz na to jakieś testy żeby zademonstrować działanie? Widzę duplikację kodu w if/else, możesz ją wyłączyć przed if-else?